src/main/java/com/delphix/masking/initializer/maskingApi/SetupDriver.java

```diff
@@ -278,6 +287,26 @@ private void handleFileFormats(List<FileFormat> fileFormats) throws IOException
         }
     }
 
+    private void handleJdbcDrivers(List<JdbcDriver> jdbcDrivers) throws ApiCallException {
+        GetJdbcDrivers getJdbcDriver = new GetJdbcDrivers();
+        apiCallDriver.makeGetCall(getJdbcDriver);
+        for (JdbcDriver jdbcDriver : jdbcDrivers) {
+            String driverFileName = "JDBC_DRIVER_" + jdbcDriver.getJdbcDriverId() + ".zip";
+            File driverFile = new File(baseFolder.resolve(driverFileName).toString());
+            // Update if driver is already exist
+            boolean isUpdate = false;
+            for (JdbcDriver existingJdbcDriver : getJdbcDriver.getJdbcDrivers()) {
+                if (existingJdbcDriver.getJdbcDriverId().equals(jdbcDriver.getJdbcDriverId())) {
+                    apiCallDriver.makePutCall(new PutJdbcDriver(jdbcDriver, driverFile));
+                    isUpdate = true;
+                }
+            }
+            if(!isUpdate) {
+                apiCallDriver.makePostCall(new PostJdbcDriver(jdbcDriver, driverFile));
+            }
```

Comment:
Using the JDBC driver id as a way to match JDBC drivers actually doesn't work properly across engines. Instead, we should be using the jdbc driver name in the file and the jdbc driver name when comparing existing jdbc drivers.

Lets say I upload an oracle jdbc driver to masking engine `A` and an mssql jdbc driver to masking engine `B`. Both of these jdbc drivers may have an ID of `1`.  If I backup engine `A` and then restore the engine `B`, then I will be overwriting the mssql driver with the oracle driver, even though they are not the same driver.

A similar example is file formats. Two different file formats may have the same id on different engines, so instead of using the file format id, we use the file format name.

```
        // Add any pre-existing file formats to the map so file metadata can be uploaded properly
        GetFileFormats getFileFormats = new GetFileFormats();
        apiCallDriver.makeGetCall(getFileFormats);
        for (FileFormat fileFormat : getFileFormats.getFileFormats()) {
            if (!this.fileFormats.containsKey(fileFormat.getFileFormatName())) {
                this.fileFormats.put(fileFormat.getFileFormatName(), fileFormat.getFileFormatId());
            }
        }
```

Please also update the exected name of the jdbc driver zip file to be `JDBC_DRIVER_<DRIVER_NAME>.zip` instead of `<ID>`.

Reply:
Thanks for pointing this issue. Fixed